Jetpack focus enable more widgets on jetpack app

ravishanker opened the pull request:
This PR adds the following widgets on Jetpack app and also dynamic widget previews for Android 12 (API 31) and above
- At a glance
- All time
- Today

Fixes #17143 

<img width=320 src="https://user-images.githubusercontent.com/990349/188815641-1ddd43e1-afa2-46e9-b197-0dcd7e411a0f.png" />



To test:

Test 1:
- Build and run Jetpack app
- Long press on Jetpack app icon in app launcher or on home screen
- Ensure it show now **Widgets** option in short cuts menu

Test 2:
- Tap on **Widgets** option
- Ensure it pops up a dialog with three widgets above; and QuickPress shortcut and Views this week widget

Test 3:
- Tap and hold any widget, then drop it on home screen
- Ensure it pops up  dialog with options for Site and Color selections
- Select a Site from popup
- Select a Color; light or dark from popup
- Tap **ADD WIDGET** button which should be enable after the selections
- Ensure the widget is added to home screen

Test 4
- Use a device or Emulator with Android 12 (API 31) and above
- Change the language to other than English like French, German or Arabic
- Observe now the widget previews now actually reflect the language

NOTE:
- App icon is defaulted to Jetpack now on dynamic previews since it is anyways Jetpack powered even on WP.


## Regression Notes
1. Potential unintended areas of impact
 Tested to make sure the widgets on WordPress app work as before and not affected by this on Jetpack app

2. What I did to test those areas of impact (or what existing automated tests I relied on)
 Tested WordPress app and Jetpack apps separately

3. What automated tests I added (or what prevented me from doing so)
 Existing and updated unit tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ravishanker committed:
Enable more widgets for JP

Enabling the following widgets for Jetpack app
- All Time
- Today
- At a glance

ravishanker committed:
Add dynamic preview for week views

Added dynamic preview for Views this week widget for API 12 (31) and above

ravishanker committed:
Add dynamic preview for All time widget

ravishanker committed:
dynamic preview for At a glance widget

ravishanker committed:
dynamic preview for Today widget

ravishanker committed:
Dynamic preview for old Views this week

ravishanker committed:
Fix lint issues

By renaming title container in preview

ravishanker committed:
Update RELEASE-NOTES.txt

ravishanker committed:
Update views widget preview

Updated Views this week widget on WP to show three rows with different colors for neutral, positive and negative change

ravishanker committed:
Merge branch 'trunk' into Jetpack-focus-Enable-more-widgets-on-Jetpack-app

irfano commented on WordPress/src/main/res/xml-v31/stats_week_views_widget_info.xml:
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<appwidget-provider
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:configure="org.wordpress.android.ui.stats.refresh.lists.widget.weeks.StatsWeekWidgetConfigureActivity"
+    android:initialKeyguardLayout="@layout/stats_widget_list_light"
+    android:initialLayout="@layout/stats_widget_list_light"
+    android:minHeight="110dp"
+    android:minResizeWidth="150dp"
+    android:minWidth="250dp"
+    android:previewLayout="@layout/stats_week_views_widget_preview"
+    android:resizeMode="horizontal"
+    android:updatePeriodMillis="600000"
+    android:widgetCategory="home_screen"
+    tools:ignore="UnusedResources">
+</appwidget-provider>
```
np: Since `stats_week_views_widget_info.xml` is used only in `jetpack/AndroidManifest.xml`, can we move `stats_week_views_widget_info.xml` to a Jetpack directory for adding this file only to the Jetpack apk.

ravishanker replied:
done 👍

irfano commented on WordPress/src/main/res/layout/stats_widget_title_dark.xml:
```diff
@@ -8,6 +8,8 @@
     android:layout_height="@dimen/stats_widget_list_item_height"
     android:background="@drawable/bg_rectangle_gray_700_top_radius_4dp"
     android:orientation="horizontal"
+    android:visibility="gone"
+    tools:visibility="visible"
```
I wonder why we make visibility gone here, then make it visible programmatically. What is the reason? 🤔

ravishanker replied:
It is for the widget previews.  The title is separate included layout along with ListView in that layout at the top.  Title is not an item row like the data.  Previews need their titles separately!

irfano reviewed (commented):
Thanks for handling this! I tested different cases. Everything works fine. 👍🏻 Some notes:
- After adding a widget, I changed the language of the device, then go back home screen. The words weren't translated until I tap the widget to open the app. This is also the same on `trunk`. I'm not sure this is related to our app or OS but I tested other Google widgets, and they don't have this problem. I think we can open a new issue with low priority for this.
- The only problem we should consider is the widget preview of "At a glance" widget. Screenshots are from Android 12, 5.7 1080x2280 emulator.

|before|after|
|-|-|
|<img src="https://user-images.githubusercontent.com/2471769/189114949-80ba140e-4457-42e1-a0df-7f717aaea167.png" height=640>|<img src="https://user-images.githubusercontent.com/2471769/189115036-d5ae849e-3861-4e57-b8c6-9139fea2a839.png" height=640>|

ravishanker commented:
> Thanks for handling this! I tested different cases. Everything works fine. 👍🏻 Some notes:
> 
>     * After adding a widget, I changed the language of the device, then go back home screen. The words weren't translated until I tap the widget to open the app. This is also the same on `trunk`. I'm not sure this is related to our app or OS but I tested other Google widgets, and they don't have this problem. I think we can open a new issue with low priority for this.

Sounds good

>     * The only problem we should consider is the widget preview of "At a glance" widget. Screenshots are from Android 12, 5.7 1080x2280 emulator.



Is the issue about cutting off or the title at the top?  I've added title at the top to be consistent with the rest of the previews.  As for cutting-off, the bottom-sheet should be in expanded state, not sure why it is half-mast on the emulator!

ravishanker committed:
Relocate xml files to Jetpack directory

Moved Jetpack only Views this week widget info xml files under Jetpack

ravishanker committed:
Move drawables to JP from WP

irfano reviewed (approved):
> Is the issue about cutting off or the title at the top? I've added title at the top to be consistent with the rest of the previews. As for cutting-off, the bottom-sheet should be in expanded state, not sure why it is half-mast on the emulator!

The issue was cutting off.  I tested again on a real device, and it worked okay. So, I'm approving the PR. 👍🏻
